### ISSUE TITLE
git-master does not build for Android x86 & x86_64

### ISSUE THREAD

**Issue by yan12125** [openssl/openssl#5643]
First, I'd like to say thank you for the awesome work in https://github.com/openssl/openssl/pull/5589. For ARM and AArch64 (ARM64), the new build system works like a charm with clang and unified headers. Here I'd like to report a minor issue: a path calculation for x86 & x86_64 is incorrect.

If I configure OpenSSL with:
```
./Configure --prefix=/usr --openssldir=/etc/ssl android-x86 no-shared -D__ANDROID_API__=21
```
The build fails at:
```
${LDCMD:-clang} -pie -pthread -Wa,--noexecstack -target i686-linux-android -gcc-toolchain /home/yen/Projects/android-ndk-r17-beta1/toolchains/i686-linux-android-4.9/prebuilt/linux-x86_64 --sysroot=/home/yen/Projects/android-ndk-r17-beta1/platforms/android-21/arch-x86 -Qunused-arguments -Wall -O3 -fomit-frame-pointer -L.   \
        -o apps/openssl apps/asn1pars.o apps/ca.o apps/ciphers.o apps/cms.o apps/crl.o apps/crl2p7.o apps/dgst.o apps/dhparam.o apps/dsa.o apps/dsaparam.o apps/ec.o apps/ecparam.o apps/enc.o apps/engine.o apps/errstr.o apps/gendsa.o apps/genpkey.o apps/genrsa.o apps/nseq.o apps/ocsp.o apps/openssl.o apps/passwd.o apps/pkcs12.o apps/pkcs7.o apps/pkcs8.o apps/pkey.o apps/pkeyparam.o apps/pkeyutl.o apps/prime.o apps/rand.o apps/rehash.o apps/req.o apps/rsa.o apps/rsautl.o apps/s_client.o apps/s_server.o apps/s_time.o apps/sess_id.o apps/smime.o apps/speed.o apps/spkac.o apps/srp.o apps/storeutl.o apps/ts.o apps/verify.o apps/version.o apps/x509.o \
         apps/libapps.a -lssl -lcrypto -ldl -pthread 
/..//bin/ld: cannot find -lgcc
/..//bin/ld: cannot find -lgcc
clang: error: linker command failed with exit code 1 (use -v to see invocation)
```
In my case, the value for ```-gcc-toolchain``` is:
```
$ANDROID_NDK/toolchains/i686-linux-android-4.9/prebuilt/linux-x86_64
```
It should be:
```
$ANDROID_NDK/toolchains/x86-4.9/prebuilt/linux-x86_64
```
I got a similar issue when building OpenSSL for Android x86_64.

I tested with Android NDK r17 beta 1, while I remember the path for x86 has been "x86-4.9" since long time ago.

**Comment by dot-asm** [comment]
It's being addressed in #5613.

**Comment by yan12125** [comment]
Thanks for the follow-up and sorry for not noticing there's already a fix (yes, there are SOOOOOO many pull requests in OpenSSL :)

Just tested with NDK r17 beta1. ARM, ARM64, x86, x86_64 worked fine. Just that OpenSSL for ARM requires adding -latomic to ./Configure. Not sure why.

Closing as this issue is already covered by #5613.
